source/Calamari/Integration/FileSystem/CalamariPhysicalFileSystem.cs

```diff
@@ -342,7 +351,15 @@ static string GetTempBasePath()
         public string CreateTemporaryDirectory()
         {
             var path = Path.Combine(GetTempBasePath(), Guid.NewGuid().ToString());
-            Directory.CreateDirectory(path);
+            return CreateDirectory(path);
+        }
+
+        public string CreateDirectory(string path)
```

Comment:
I wanted to use CalamariPhysicalFileSystem for it's cross platform logic it was already being used in tests.